src/snapred/backend/data/LocalDataService.py

```diff
@@ -590,21 +592,17 @@ def writeCalibrationWorkspaces(self, record: CalibrationRecord):
         for wsName in workspaces.pop(wngt.DIFFCAL_OUTPUT, []):
             # Rebuild the filename to strip any "iteration" number:
             #   * WARNING: this workaround does not work correctly if there are multiple workspaces of each "unit" type.
+            ext = "tar"
+            nameBuilder = wng.diffCalOutput().runNumber(record.runNumber).version(record.version)
             if wng.Units.TOF.lower() in wsName:
-                filename = Path(
-                    wng.diffCalOutput().unit(wng.Units.TOF).runNumber(record.runNumber).version(record.version).build()
-                    + ".nxs"
-                )
+                filename = Path(f"{nameBuilder.unit(wng.Units.TOF).build()}.{ext}")
```

Comment:
Did you mean to revert the builder here?

Reply:
I didnt revert it, I'm using the builder for its intermediates here rather than constructing it all from scratch in both tof and dsp case.

Side note but I realize the units are wrong for dsp